protostar/configuration_file/configuration_file.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+from typing import List, Optional, Union
+
+from typing_extensions import Protocol
+
+from protostar.protostar_exception import ProtostarException
+from protostar.utils.protostar_directory import VersionType
+
+PrimitiveTypesSupportedByConfigurationFile = Union[str, int, bool]
+
+
+class ConfigurationFile(Protocol):
```

Comment:
what about writing `protostar.toml` ?

Reply:
I am working on a different branch. I added a method that returns `ConfigurationWriteFriendlyRepresentation` that's going to be used by `ProtostarTOMLWriter`.

Reply:
> I am working on it in a different branch/PR.

👍🏻 



> ConfigurationFileV1 will be responsible for writing data. ConfigurationFileV1 should communicate with the ProtostarTOMLReader/Writer in format independent way.

See my other comments why I don't like this.